manifest: mbedtls: update to 3.6.4

ceolin opened the pull request:
Pull in Mbed TLS 3.6.4, updated from 3.6.3.

More details regarding the Mbed TLS 3.6.4 release at: https://github.com/Mbed-TLS/mbedtls/releases/tag/mbedtls-3.6.4

ceolin committed:
manifest: mbedtls: update to 3.6.4

Pull in Mbed TLS 3.6.4, updated from 3.6.3.

More details regarding the Mbed TLS 3.6.4 release at:
https://github.com/Mbed-TLS/mbedtls/releases/tag/mbedtls-3.6.4

Signed-off-by: Flavio Ceolin <flavio@hubblenetwork.com>

tomi-font reviewed (commented):
Please also update the release notes (there was already an entry for 3.6.2 to 3.6.3) in a separate commit.

danieldegrasse committed:
doc: releases: 4.2: add notes for mbedtls 3.6.4 update

Add release notes for mbedtls 3.6.4 update, including CVEs resolved with
this release.

Signed-off-by: Daniel DeGrasse <ddegrasse@tenstorrent.com>

tomi-font reviewed (approved):
Thanks @ceolin and @danieldegrasse!
